hawkular-dmr-client/src/main/java/org/hawkular/dmr/api/OperationBuilder.java

```diff
@@ -0,0 +1,446 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.StringTokenizer;
+
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.Property;
+
+/**
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class OperationBuilder implements DmrConstants, SubsystemDatasourceConstants, SubsystemLoggingConstants {
+    private static final MsgLogger log = DmrApiLoggers.getLogger(OperationBuilder.class);
+
+    public abstract static class AbstractOperationBuilder<T extends AbstractOperationBuilder<?, R>, //
+    R extends OperationResult<?>> {
+        protected final ModelNode baseNode = new ModelNode();
+
+        public ModelNode build() {
+            return baseNode;
+        }
+
+        protected R createResult(ModelNode request, ModelNode result) {
+            return (R) new OperationResult(request, result);
+        }
+
+        public R execute(ModelControllerClient client) {
+            ModelNode request = build();
+            ModelNode result;
+            try {
+                result = client.execute(request);
+            } catch (IOException e) {
+                throw new DmrApiException(e);
+            }
+            log.tracef("Executed [%s] built by [%s] with result [%s]", request, getClass().getName(), result);
+            return createResult(request, result);
+        }
+
+    }
+
+    public abstract static class AbstractSingleOperationBuilder //
+    <T extends AbstractSingleOperationBuilder<?, R>, R extends OperationResult<?>>
+            extends AbstractOperationBuilder<T, R> {
+        protected final BatchOperationBuilder<BatchOperationBuilder<?>> batch;
+
+        private AbstractSingleOperationBuilder(BatchOperationBuilder<BatchOperationBuilder<?>> batch,
+                String operationName) {
+            super();
+            this.batch = batch;
+            this.baseNode.get(OP).set(operationName);
+        }
+
+        public AddressBuilder<AddressBuilder<?, T>, T> address() {
+            return new AddressBuilder<AddressBuilder<?, T>, T>((T) this);
+        }
+
+        public T address(ModelNode addess) {
+            baseNode.get(ADDRESS).set(addess);
+            return (T) this;
+        }
+
+        public BatchOperationBuilder<?> parentBuilder() {
+            return batch.operation(build());
+        }
+
+    }
+
+    public static class AddOperationBuilder<T extends AddOperationBuilder<?>>
+            extends AbstractSingleOperationBuilder<T, OperationResult<?>> {
+
+        private AddOperationBuilder(BatchOperationBuilder<BatchOperationBuilder<?>> batch) {
+            super(batch, ADD);
+        }
+
+        /**
+         * If {@code value == null}, no entry is added. Otherwise, the attribute will be added.
+         *
+         * @param name
+         * @param value
+         * @return
```

Comment:
Holly yes, I know you see everything, which is one of the reasons I have put [Do not merge yet] on this PR. I should turn that stupid validator on (in my workspace at least) not to bother you with such stupidities. But thinking of it once again, your care for these nits might also be interpreted as a sign that you like the underlying code(?)